scipy/signal/signaltools.py

```diff
@@ -777,6 +777,88 @@ def convolve(in1, in2, mode='full', method='auto'):
     return correlate(volume, _reverse_and_conj(kernel), mode, 'direct')
 
 
+def circular_convolve(in1, in2, period):
+    """
+    Circular convolution of two 1-dimensional arrays.
+
+    Circular convolve `in1` and `in2` with given `period`.
+
+    Parameters
+    ----------
+    in1 : array_like, 1-D
+        First input.
+    in2 : array_like, 1-D
+        Second input. Should have the same number of dimensions as `in1`.
```

Comment:
Ideally these functions would operate along a dimension with an `axis` argument. But for now you can just remove the statement about having the same number of dimensions (both must be 1).

You should probably add some checks that it is 1D, or say that it will be `.ravel()`ed (and do it).